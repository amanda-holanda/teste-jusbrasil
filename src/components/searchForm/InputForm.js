export const InputForm = () => {
  return (
    <input
      className="inputSearch"
      type="text"
      name="numeroProcesso"
      placeholder="1234567-88.2020.8.10.1234"
      required
    />
  );
};
