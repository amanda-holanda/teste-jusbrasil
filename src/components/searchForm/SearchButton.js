export const SearchButton = () => {
  return (
    <button className="searchBtn" type="submit">
      Buscar
    </button>
  );
};
