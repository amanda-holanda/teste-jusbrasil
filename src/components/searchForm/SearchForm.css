.searchForm {
  display: flex;
  justify-content: center;
  align-items: center;  
  width: 85vw;
  flex-direction: column;
  background-color: lightgray;
  margin: 0% 5% 3%;  
  height: 30vh;
  border-radius: 5px;
  overflow: hidden;
}

.searchTitle {
  margin: 1% 0 1%;
  font-size: 1.8em;
  text-align: center;
}

.searchParagraph {
  margin: 0 0 1%;
  text-align: center;
}

.inputButtonContainer {
  display: flex;  
  flex-direction: column;
  width: 100%;
  justify-content: center;
  align-items: center;
  margin: 3% 0 1%;
}

.labelSearch {
  display: flex;
  justify-content: center;
}

.inputSearch {
  border: 1px solid gray;
  height: 6vh;
  width: 60vw;
  font-size: 1em;
  padding: 0 0 0 3%;
  background-color: white;
  border-radius: 5px;
  color: rgb(134, 132, 132);
  align-self: center;
}

.searchBtn {
  height: 6vh;
  font-weight: bolder;
  text-transform: uppercase;
  background: rgb(46, 45, 45);
  color: white;
  width: 25vw;
  font-size: 1em;
  border-radius: 5px;
  border: 0;
  margin: 2% 0 0;
}

.searchBtn:hover {
  cursor: pointer;
  opacity: 0.5;
}

@media screen and (min-width: 768px) {
  .inputButtonContainer {
    flex-direction: row;
    gap: 2vw;
    margin: 1% 0 1%;
  }

  .inputSearch {
    max-width: 30vw;
  }  

  .searchBtn {
    margin: 0;
    width: 120px;
  }
}

@media screen and (min-width: 1024px) {
  .inputSearch {
    max-width: 18vw;
  }
}
