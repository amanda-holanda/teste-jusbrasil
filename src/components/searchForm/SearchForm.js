import "./SearchForm.css";
import { InputForm } from "./InputForm";
import { SearchButton } from "./SearchButton";

export const SearchForm = () => {
  return (
    <form action="/display" className="searchForm">
      <h1 className="searchTitle">Consulta Processual</h1>
      <p className="searchParagraph">
        Consultar processos por n√∫mero em um clique
      </p>
      <div className="inputButtonContainer">
        <label className="labelSearch" htmlFor="numeroProcesso">
          <InputForm />
        </label>
        <SearchButton />
      </div>
    </form>
  );
};
