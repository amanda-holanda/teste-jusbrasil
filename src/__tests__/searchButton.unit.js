import { render, screen } from "@testing-library/react";
import { SearchButton } from "../components/searchForm/SearchButton";

describe("<SearchButton />", () => {
  it("Renderiza um botÃ£o corretamente", () => {
    render(<SearchButton>Buscar</SearchButton>);
    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();
    expect(button).toMatchSnapshot();
  });
});
