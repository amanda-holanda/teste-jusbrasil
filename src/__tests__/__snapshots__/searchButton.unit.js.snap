// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchButton /> Renderiza um bot√£o corretamente 1`] = `
<button
  class="searchBtn"
  type="submit"
>
  Buscar
</button>
`;
